File: src/CLI.Tests/TestFixtures/Models/GenericModel.cs
namespace TestFixtures.Models;

/// <summary>
/// A generic model for testing generic type handling.
/// </summary>
public class GenericModel<T>
{
    public T Value { get; set; } = default!;
    public List<T> Items { get; set; } = new();
}

/// <summary>
/// Model that inherits from a generic base class.
/// </summary>
public class StringModel : GenericModel<string>
{
    public string Description { get; set; } = string.Empty;
}

File: src/ItemTemplates/Empty/GenericModel.ts

module $rootnamespace$ {

    // $Classes/Enums/Interfaces(filter)[template][separator]
    // filter (optional): Matches the name or full name of the current item. * = match any, wrap in [] to match attributes or prefix with : to match interfaces or base classes.
    // template: The template to repeat for each matched item
    // separator (optional): A separator template that is placed between all templates e.g. $Properties[public $name: $Type][, ]

    // More info: http://frhagn.github.io/Typewriter/

    
    export class GenericModel {
        
        // $LoudName
        public value: T = new T();
        // $LoudName
        public items: T[] = [];
    }
    export class StringModel {
        
        // $LoudName
        public description: string = "";
    }
}